run-build-functions.sh

```diff
@@ -779,6 +817,56 @@ restore_cwd_cache() {
   move_cache "$NETLIFY_CACHE_DIR/$1" "$PWD/$1" "restoring cached $2"
 }
 
+#
+# Restores node_modules dirs cached for js workspaces
+# See https://github.com/netlify/pod-workflow/issues/139/ for more context
+#
+# Expects:
+# $@ each argument should be a package location relative to the repo's root
+restore_js_workspaces_cache() {
+  local locations=("$@")
+  local cache_dir="$NETLIFY_CACHE_DIR/js-workspaces"
+
+  # Retrieve each workspace node_modules
+  for location in "${locations[@]}"; do
+    move_cache "$cache_dir/$location/node_modules" "$NETLIFY_REPO_DIR/$location/node_modules" "restoring workspace $location node modules"
+  done
+  # Retrieve hoisted node_modules
+  move_cache "$cache_dir/node_modules" "$NETLIFY_REPO_DIR/node_modules" "restoring workspace root node modules"
+  # Keep a record of the workspaces in the project in order to cache them later
+  NETLIFY_JS_WORKSPACE_LOCATIONS=$(printf '%s\n' "${locations[@]}")
```

Comment:
Would it work to keep a record of `locations` directly instead of using `NETLIFY_JS_WORKSPACE_LOCATIONS` as an intermediate variable?
Like this, `cache_js_workspaces` could use it directly, without having to use `mapfile`.

Reply:
Good point @ehmicky! Addressed via https://github.com/netlify/build-image/commit/32f7b0df2c181f9dfeb4f654b1115ead254e46cf.